com.sap.cloud.lm.sl.mta/src/main/java/com/sap/cloud/lm/sl/mta/builders/v1/ModuleDependenciesCollector.java

```diff
@@ -0,0 +1,74 @@
+package com.sap.cloud.lm.sl.mta.builders.v1;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+import com.sap.cloud.lm.sl.mta.handlers.v1.DescriptorHandler;
+import com.sap.cloud.lm.sl.mta.model.v1.DeploymentDescriptor;
+import com.sap.cloud.lm.sl.mta.model.v1.Module;
+
+public class ModuleDependenciesCollector {
+
+    private DescriptorHandler handler;
+    protected DeploymentDescriptor descriptor;
+    protected Set<String> seenModules;
+
+    public ModuleDependenciesCollector(DeploymentDescriptor descriptor, DescriptorHandler handler) {
+        this.descriptor = descriptor;
+        this.handler = handler;
+    }
+
+    public Set<String> collect(Module module) {
+        seenModules = new HashSet<>();
+        return getRequiredModulesRecursively(module);
+    }
+
+    private Set<String> getRequiredModulesRecursively(Module module) {
+        if (visited(module)) {
+            return Collections.emptySet();
+        }
+
+        return collectRequiredModulesRecursively(module);
+    }
+
+    private boolean visited(Module module) {
+        if (seenModules.contains(module.getName())) {
+            return true;
+        }
+        seenModules.add(module.getName());
+        return false;
+    }
+
+    private Set<String> collectRequiredModulesRecursively(Module module) {
+        Set<String> requiredModules = new LinkedHashSet<>();
+        for (String dependency : getDependencies(module)) {
+            Module requiredModule = findModuleSatisfyingDependency(dependency);
+            if (isModuleValidRequirement(module, requiredModule)) {
```

Comment:
I do not understand this: "isModuleValidRequirement"? is there any chance we could rename it?